components/views.py

```diff
@@ -1,67 +1,63 @@
     Provides get and post method handlers.
     """
-
     queryset = Component.objects.all().order_by("pk")
     serializer_class = ComponentListSerializer
-
-    def list(self, request):
-        queryset = self.get_queryset()
-        serializer = ComponentListSerializer(queryset, many=True)
-        return Response(serializer.data, status=status.HTTP_200_OK)
+    filter_backends = [ObjectPermissionsFilter, ]
 
 
-class ComponentDetailView(APIView):
+class ComponentDetailView(generics.RetrieveAPIView):
     """
     Use for read or update endpoints to represent a single model instance.
     Provides get, put, and patch method handlers.
     """
+    queryset = Component.objects.all()
+    serializer_class = ComponentSerializer
 
-    def get(self, request, pk):
-        component = get_object_or_404(Component, pk=pk)
-        serializer = ComponentSerializer(component, context={"request": request})
-        return Response(serializer.data, status=status.HTTP_200_OK)
 
+class ComponentListSearchView(generics.ListAPIView):
+    queryset = Component.objects.all().order_by("pk")
+    filterset_class = ComponentFilter
+    filter_backends = [ObjectPermissionsFilter, filters.DjangoFilterBackend, ]
+    serializer_class = ComponentListSerializer
 
-class ComponentListSearchView(APIView):
-    def get(self, request, *args, **kwargs):
+    def list(self, request, *args, **kwargs):
+        queryset = self.filter_queryset(self.get_queryset())
         page_number = self.request.query_params.get("page", default=1)
 
-        filtered_qs = ComponentFilter(
-            self.request.GET,
-            queryset=Component.objects.all().order_by("pk"),
-        ).qs
-
-        paginator = Paginator(filtered_qs, 20)
+        paginator = Paginator(queryset, 20)
         try:
             page_obj = paginator.page(page_number)
         except PageNotAnInteger:
             page_obj = paginator.page(1)
         except EmptyPage:
             page_obj = paginator.page(paginator.num_pages)
 
-        serializer_class = ComponentListSerializer
-        serializer = serializer_class(page_obj, many=True)
-
+        serializer = self.get_serializer(page_obj, many=True)
         response = serializer.data
         response.append({"total_item_count": paginator.count})
+
         return Response(response, status=status.HTTP_200_OK)
 
 
-class ComponentTypeListView(APIView):
-    def get(self, *args, **kwargs):
-        type_list = Component.objects.order_by().values_list("type").distinct()
-        return Response(type_list, status=status.HTTP_200_OK)
+class ComponentTypeListView(generics.ListAPIView):
+    queryset = Component.objects.order_by().values_list("type").distinct()
+    filter_backends = [ObjectPermissionsFilter, ]
+
+    def list(self, request, *args, **kwargs):
+        queryset = self.filter_queryset(self.get_queryset())
+
+        return Response(queryset, status=status.HTTP_200_OK)
```

Comment:
This isn't ideal, and we should probably revisit this at some point. Is the purpose just to get a set of available component types?